sql: strip tenant ID from key before calling IndexKeyEquivSignature

asubiotto opened the pull request:
IndexKeyEquivSignature is written to expect no tenant ID prefix. ReadIndexKey
was incorrectly calling this function with a key including a tenant ID prefix.
This commit strips the prefix.

Release note: None

Fixes #50048 
Fixes #50050

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52562)
<!-- Reviewable:end -->

asubiotto commented:
cc @jordanlewis

jordanlewis commented:
LGTM. Another thing that hopefully we can remove once we remove interleaved tables.

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52562#-:-MENglNF18Nzi-OFmMhC:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @tbg)

___
*[pkg/sql/logictest/testdata/logic_test/cascade, line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52562#-MENgLRW8L50RF-qW41L:-MENgLRW8L50RF-qW41M:b6i8m5h) ([raw file](https://github.com/cockroachdb/cockroach/blob/fdf54ebc69c7d975f8f1dfbf28ced79ddeb83ee4/pkg/sql/logictest/testdata/logic_test/cascade#L2)):*
> ```
> # LogicTest: !3node-tenant(50048)
> ```

nit: stray line
___
*[pkg/sql/row/fetcher.go, line 779 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52562#-MENgOl27ESVHkDw-9dl:-MENgOl27ESVHkDw-9dm:b6565fk) ([raw file](https://github.com/cockroachdb/cockroach/blob/39a8f24221665709fbc2d0d96efea9fe8b1f2b45/pkg/sql/row/fetcher.go#L779)):*
> ```Go
> 	initialKey := key
> 
> 	if !rf.codec.ForSystemTenant() {
> ```

You shouldn't need to guard this on the tenant – strip tenant prefix is a no-op for the system tenant.
___
*[pkg/sql/sqlbase/index_encoding.go, line 1238 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52562#-MENgi-x7p5I5WVmlQzZ:-MENgi-x7p5I5WVmlQz_:b-oe8tsj) ([raw file](https://github.com/cockroachdb/cockroach/blob/39a8f24221665709fbc2d0d96efea9fe8b1f2b45/pkg/sql/sqlbase/index_encoding.go#L1238)):*
> ```Go
> }
> 
> // IndexKeyEquivSignature parses an index key if and only if the index
> ```

nit: consider rewrapping this comment.


<!-- Sent from Reviewable.io -->

Commit:
sql: strip tenant ID from key before calling IndexKeyEquivSignature

IndexKeyEquivSignature is written to expect no tenant ID prefix. ReadIndexKey
was incorrectly calling this function with a key including a tenant ID prefix.
This commit strips the prefix.

Release note: None

asubiotto commented:
Done. TFTRs

bors r=nvanbenschoten,jordanlewis
